File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.py
def solution(n,a,b):
	if a<b:
		answer = 1
		while a%2!=1 or (a+1!=b):
			answer+=1
			a=((a-1)//2+1)
			b=((b-1)//2+1)
	else:
		answer = 1
		while b%2!=1 or (b+1!=a):
			answer+=1
			a=((a-1)//2+1)
			b=((b-1)//2+1)
	return answer

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution{
    public int solution(int n, int a, int b){
        // int answer = 0;
        // if(a<b){
        //     answer=match(a,b);
        // }else{
        //     answer=match(a,b);
        // }

        return a<b ? match(a,b) : match(b,a);
    }
    public static int match(int a,int b){// b가 더 크다. 
        int count = 1;
        while(a+1!=b || a%2!=1){ // a가 짝수이면
            count++;
            if(a%2==1){// a 가 홀수 일 때
                a+=1;
            }
            if(b%2==1){
                b+=1;
            }
            a=a/2;
            b=b/2;
        }
        return count;
        
    }
}



